xla/hlo/transforms/collectives/BUILD

```diff
@@ -351,6 +351,52 @@ xla_cc_test(
     ],
 )
 
+cc_library(
+    name = "collective_permute_combiner",
+    srcs = ["collective_permute_combiner.cc"],
+    hdrs = ["collective_permute_combiner.h"],
+    deps = [
+        "//xla:shape_util",
+        "//xla:status_macros",
+        "//xla:xla_data_proto_cc",
+        "//xla/hlo/ir:hlo",
+        "//xla/hlo/pass:hlo_pass",
+        "//xla/hlo/utils:hlo_query",
+        "//xla/hlo/utils:hlo_sharding_util",
+        "//xla/service:collective_permute_key",
+        "//xla/service:collective_combiner_utils",
+        "//xla/service:hlo_domain_map",
+        "@com_google_absl//absl/container:flat_hash_set",
+        "@com_google_absl//absl/functional:function_ref",
+        "@com_google_absl//absl/log",
+        "@com_google_absl//absl/status",
+        "@com_google_absl//absl/status:statusor",
+        "@com_google_absl//absl/strings",
+        "@com_google_absl//absl/types:span",
+        "@tsl//tsl/platform:errors",
+        "@tsl//tsl/platform:statusor",
+    ],
+)
+
+xla_cc_test(
+    name = "collective_permute_combiner_test",
+    srcs = ["collective_permute_combiner_test.cc"],
+    deps = [
+        ":collective_permute_combiner",
+        "//xla:literal_util",
+        "//xla:shape_util",
+        "//xla:xla_data_proto_cc",
+        "//xla/hlo/ir:hlo",
+        "//xla/hlo/testlib:hlo_hardware_independent_test_base",
+        "//xla/hlo/utils:hlo_matchers",
+        "@com_google_absl//absl/log",
+        "@com_google_absl//absl/log:check",
+        "@com_google_googletest//:gtest_main",
+        "@tsl//tsl/platform:statusor",
+        "@tsl//tsl/platform:test_main",
```

Comment:
This causes a build failure internally.
```
ld: error: duplicate symbol: main
>>> defined at gunit_main.cc
>>>            blaze-out/k8-opt/bin/testing/base/internal/_objs/gunit_main_src/gunit_main.pic.o:(main)
>>> defined at test_main.cc
>>>            blaze-out/k8-opt/bin/third_party/tensorflow/compiler/xla/tsl/platform/_objs/test_main/test_main.pic.o:(.text+0x0)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
I believe you can just remove this dependency since the target already depends on gtest_main.

Reply:
@terryysun We fixed it internally and have merged the PR. No action is needed from your side. Sorry again for the delay!